server/src/rate_limit/rate_limiter.rs

```diff
@@ -79,7 +79,7 @@ impl RateLimiter {
   }
 
   #[allow(clippy::float_cmp)]
-  fn check_rate_limit_full(
+  pub(super) fn check_rate_limit_full(
```

Comment:
I've never seen this before, what does adding super do? The only callers I can find for this function are the 3 callers in this same file.

Reply:
this makes `check_rate_limit` visible to the module directly above this one. in this case, that's mod.rs